app/models/region.rb

```diff
@@ -54,6 +54,23 @@ def log_payload
     attrs.symbolize_keys
   end
 
+  def self.sources
+    includes(:source).map(&:source)
+  end
+
+  def syncable_patients
```

Comment:
I'll bench this in isolation, but I think this will be a lot slower than before.